docker/indexer/indexer.go

```diff
@@ -34,67 +33,70 @@ var (
 	configsBucket = flag.String("configs", "", "bucket containing the textproto configs")
 	reposBucket   = flag.String("repos", "", "bucket for storing the repository data")
 	projectID     = flag.String("project_id", "", "the gcp project ID")
+	worker        = flag.Bool("worker", false, "makes this a worker node reading from pubsub to process the data")
+	pubsubTopic   = flag.String("topic", "", "sets the pubsub topic to publish to or to read from")
+	subName       = flag.String("subscription", "", "sets the pubsub subscription name for workers")
 )
 
 func main() {
 	flag.Parse()
 
 	ctx := context.Background()
+
+	psCl, err := pubsub.NewClient(ctx, *projectID)
+	if err != nil {
+		log.Exitf("failed to initialize pubsub client: %v", err)
+	}
+	defer psCl.Close()
+
+	topic := psCl.Topic(*pubsubTopic)
```

Comment:
Move this below into the non-worker part since it's not needed by the worker? (ideally into a `runController` function which I suggested below)